zanata-war/src/main/java/org/zanata/events/DocStatsEvent.java

```diff
@@ -0,0 +1,58 @@
+package org.zanata.events;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.zanata.common.ContentState;
+
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+import javax.annotation.Nullable;
+
+/**
+ * @author Alex Eng <a href="mailto:aeng@redhat.com">aeng@redhat.com</a>
+ */
+@AllArgsConstructor
+@Getter
+@EqualsAndHashCode
+public class DocStatsEvent {
+    private final DocumentLocaleKey key;
+
+    private final Long projectVersionId;
+
+    /**
+     * Updated content states with word counts
+     */
+    private final Map<ContentState, Integer> contentStates;
```

Comment:
- [ ] I would be tempted to call this something like wordDeltasByState. The ContentState bit is part of the type, but it's not automatically obvious what the Integers are for.